src/lib.rs

```diff
@@ -444,43 +466,116 @@ pub struct QuicStatisticsMisc {
     pub key_update_count: u32,
 }
 
+bitfield! {
+    #[repr(C)]
+    #[derive(Serialize, Deserialize, Clone, Copy)]
+    pub struct QuicStatisticsBitfields(u32);
+    //impl Debug; // TODO Debug causing panic when fmt is called
+    // The fields default to u32
+    version_negotiation, _: 1, 0;
+    stateless_retry, _: 1, 1;
+    resumption_attempted, _: 1, 2;
+    resumption_succeeded, _: 1, 3;
+}
+
+impl fmt::Debug for QuicStatisticsBitfields {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_fmt(format_args!("{:#06x}", &self.0))
+    }
+}
+
+/// A helper struct for accessing connection statistics
 #[repr(C)]
 #[derive(Serialize, Deserialize, Debug, Copy, Clone)]
+pub struct QuicStatisticsV2 {
+    correlation_id: u64,
+    pub flags: QuicStatisticsBitfields,
+    /// In microseconds
+    pub rtt: u32,
+    /// In microseconds
+    pub min_rtt: u32,
+    /// In microseconds
+    pub max_rtt: u32,
+
+    pub timing_start: u64,
+    /// Processed all peer's Initial packets
+    pub timing_start_flight_end: u64,
+    /// Processed all peer's Handshake packets
+    pub timing_handshake_fligh_end: u64,
+
+    /// Sum of TLS payloads
+    pub handshake_client_flight_1_bytes: u32,
+    /// Sum of TLS payloads
+    pub handshake_server_flight_1_bytes: u32,
+    /// Sum of TLS payloads
+    pub handshake_client_flight_2_bytes: u32,
+
+    /// Current path MTU.
+    pub send_path_mtu: u16,
+    /// QUIC packets; could be coalesced into fewer UDP datagrams.
+    pub send_total_packets: u64,
+    pub send_retransmittable_packets: u64,
+    pub send_suspected_lost_packets: u64,
+    /// Actual lost is (suspected_lost_packets - spurious_lost_packets)
+    pub send_spurious_lost_packets: u64,
+    /// Sum of UDP payloads
+    pub send_total_bytes: u64,
+    /// Sum of stream payloads
+    pub send_total_stream_bytes: u64,
+    /// Number of congestion events
+    pub send_congestion_count: u32,
+    /// Number of persistent congestion events
+    pub send_persistent_congestion_count: u32,
+
+    /// QUIC packets; could be coalesced into fewer UDP datagrams.
+    pub recv_total_packets: u64,
+    /// Packets where packet number is less than highest seen.
+    pub recv_reordered_packets: u64,
+    /// Includes DuplicatePackets.
+    pub recv_dropped_packets: u64,
+    pub recv_duplicate_packets: u64,
+    /// Sum of UDP payloads
+    pub recv_total_bytes: u64,
+    /// Sum of stream payloads
+    pub recv_total_stream_bytes: u64,
+    /// Count of packet decryption failures.
+    pub recv_decryption_failures: u64,
+    /// Count of receive ACK frames.
+    pub recv_valid_ack_frames: u64,
+
+    pub key_update_count: u32,
+    // N.B. New fields must be appended to end
```

Comment:
Considering this will never be updated separate from the main API, these type of comments aren't necessary IMO.